### ISSUE TITLE
Mapping network drive to \\wsl$\path does not work

### ISSUE THREAD

**Issue by jonitis** [Microsoft/WSL#3996]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
10.0.18362.53

* In Windows explorer write address "\\wsl$", select the displayed "Ubuntu" icon and select "Map network drive...". Try to map preselected path "Ubuntu" (grayed out, not editable) to driver letter W:. It will fail with "Windows cannot access Ubuntu" window.

Alternatively in command path try:
```
net use w: \\wsl$\Ubuntu\
System error 67 has occurred.

The network name cannot be found.
```

* What's wrong / what should be happening instead:
Would be nice to be able to map drive letter to \\wsl$\distro_path. If this is not yet possible then Explorer should not display "Map network drive..." menu entry if it is known to fail

**Comment by Biswa96** [comment]
Fixed in 18860 (Skip Ahead). See issue #3854

**Comment by kapitanluffy** [comment]
I tried mapping a directory from WSL and it does not work

![image](https://user-images.githubusercontent.com/485799/56881308-9808e500-6a92-11e9-82f7-07646fadf108.png)

**Comment by Biswa96** [comment]
See the previous linked issue.

**Comment by kapitanluffy** [comment]
@Biswa96 ok. I should wait for build 18860 then? My current build is 18362.53

Also, when I try to map using `\\wsl$\Ubuntu\` only, it works. Network locations work too.

**Comment by drillyme** [comment]
> I tried mapping a directory from WSL and it does not work
> 
> ![image](https://user-images.githubusercontent.com/485799/56881308-9808e500-6a92-11e9-82f7-07646fadf108.png)

hi there,

build 22000
same issue, thou just adding \wsl$<distro-name> works fine

**Comment by cavo789** [comment]
Hello

I had the same problem today: impossible to map the distribution; here is the solution: https://www.reddit.com/r/bashonubuntuonwindows/comments/hos80y/comment/fxleabf

TLDR: we first need to open the distribution (Ubuntu in my case), to do this, from you Windows start menu, click on the Ubuntu icon.  Then the map network drive will work from File Explorer.

**Comment by jfritz** [comment]
Hello. Same problem encountered here just today as @drillyme 's error message.

I am trying to map to a network drive because JetBrains' PyCharm debugger has difficulty showing current line-of-execution while debugging these files in a virtual wsl drive.

I imagine other programs have compatibility problems with operating on files in the wsl virtual drive as well.

**Comment by Grarie** [comment]
@cavo789 This is not the problem. The distribution is opened.
I can add \\wsl$\debian or \\wsl.localhost\debian but cannot add \\wsl$\ssh-debian\home\username
So no path can be added. Strange.

Windows 11 build:  10.0.22572.0 (22572.1.amd64fre.ni_release.220304-1536)
